hashicorp/dev-portal
# feat: stub in /hcp/docs content

## 🔗 Relevant links

- [Preview link - /hcp/docs][/hcp/docs] 🔎
- [Asana task - Determine if HCP needs additional engineering work to migrate][task-explore] 🎟️
- [Asana task - Migrate HCP docs][task-migrate-hcp-docs] 🎟️

## 🗒️ What

Adds `/hcp/docs` routes.

## 🤷 Why

To explore potential rough edges in integrating HCP documentation into the HashiCorp Developer site.

## 🛠️ How

Adds `src/pages/hcp/docs`, similar to existing docs routes for other products.

## 🧪 Testing

- Visit [the /hcp/docs route][/hcp/docs], and ensure routes render as expected
    - Sidebar and sidecar contents should look and function as expected
    - The mobile menu should contain the right links and panels, and work as expected
    - Images should load
    - Search should work, and should search within HCP documentation only

## Other notes

- **Redirects have not been added, yet**. The content source repository is private, and we're still working on potential solutions to address this issue. Intent is to handle the redirects concerns outside of this PR. Relevant links:
    - 🎟️ [[redirects] load redirects from private repositories](https://app.asana.com/0/1202097197789424/1202532915796679/f)
    - We may also consider frontmatter-style redirects for docs files, like Learn does. Context in 🎟️ [Tutorials Redirects Property -> `old_filenames`](https://app.asana.com/0/1202097197789424/1201956804128029/f)
    - Note as well that existing dev-dot routes do not redirect as they do on dot-io sites. This is another reason why it felt reasonable to defer this work outside this PR. See 🎟️ [Get remotely sourced redirects working for dev-portal UI routes](https://app.asana.com/0/1201987349274776/1201662082096106/f)
    - As a fallback solution here, we could move the redirects.js file we need from the cloud.hashicorp.com repository into this hashicorp/dev-portal repository. This would come with author visibility concerns, but would at least "work" in the most basic sense of the word.
- **Only the `/docs` route has been stubbed in**. This is the only "MDX content" route on <https://cloud.hashicorp.com>. But, <https://cloud.hashicorp.com> has many other pages. The assumption at this time is that we don't plan to migrate those pages into the HashiCorp Developer site.
- **`Edit on GitHub` link is still shown**. This should likely be removed for HCP docs. However, there's currently no way to do this without changing the interface for `SidebarSidecarLayout` server utilities. Intent is to defer this tweak until after we've refactored that layout to extract docs-view specific logic.

[task-migrate-hcp-docs]: https://app.asana.com/0/1201987349274776/1201650734121659/f
[task-explore]: https://app.asana.com/0/1202097197789424/1202371394012663/f
[/hcp/docs]: https://dev-portal-git-zsstub-hcp-docs-hashicorp.vercel.app/hcp/docs
